server/src/database.rs

```
@@ -1091,6 +1168,42 @@ struct DatabaseStateDatabaseObjectStoreFound {
 }
 
 impl DatabaseStateDatabaseObjectStoreFound {
+    /// Load owner info from object storage and verify it matches the current owner
+    async fn advance(
+        &self,
+        shared: &DatabaseShared,
+    ) -> Result<DatabaseStateOwnerInfoLoaded, InitError> {
+        let raw_owner_info = self
+            .iox_object_store
+            .get_owner_file()
+            .await
+            .context(LoadingOwnerInfo)?;
+
+        let owner_info = generated_types::server_config::decode_database_owner_info(raw_owner_info)
```

should we ignore errors if the existing `owner` file is not found? It seems like being able to delete a bad file would be a good way to get a database back up and running

---

I think deleting a bad file and then issuing an "adopt" API request would be a better way to get a database back up and running. A server thinking it owns a database and the database missing the owner file would indicate some sort of inconsistency, and if we ignore that inconsistency, then the owner file isn't serving as much of a check on the state's correctness.

---

That is a good point -- I think the key thing is to make sure there  is some way to get the system out of trouble. Your proposed API sounds like a good one. 👍